### PR TITLE
chore: Brillig BinaryFieldOp simplification, rename resolve_foreign_call_input

### LINKED ISSUE #306: Consider renaming `resolve_foreign_call_input` in Brillig VM
### Problem

Comment regarding this is [here](https://github.com/noir-lang/acvm/pull/152#discussion_r1200937418)

This method does not seem to have anything to do with foreign calls in isolation, so a method stating what it is doing would be better.

### Proposed solution

- Add doc comment to function.
- Consider renaming it to something more appropriate

### Alternatives considered

_No response_

### Additional context

_No response_

### Submission Checklist

- [X] Once I hit submit, I will assign this issue to the Project Board with the appropriate tags.

### PR DESCRIPTION
**Pull request by ludamad**
Resolves #306 #326

# Description

Various fixup changes. Includes a change to try and catch misuses of registers (by having an explicit uninitialized value)

## Summary of changes

Rename resolve_foreign_call_input for #306 
chore: Remove Field < and <= support from Brillig for #326 

## Dependency additions / changes

N/A

## Test additions / changes

Some minor test simplification as field LTE was no longer available

# Checklist

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [ ] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [ ] I have reviewed the changes on GitHub, line by line.
- [ ] I have ensured all changes are covered in the description.

# Additional context

(If applicable.)

### CODE REVIEW

File: brillig_vm/src/lib.rs [hunk 1]
```diff
@@ -471,6 +464,7 @@ mod tests {
 
     #[test]
     fn cmp_binary_ops() {
+        let bit_size = 32;
```
Comment (ludamad) [inline brillig_vm/src/lib.rs:467]: Previously this was not correct if both elements are modulo before the result (which I assume the intent is - to first cast to N bit integers)

File: brillig_vm/src/lib.rs [hunk 2]
```diff
@@ -300,14 +306,6 @@ impl VM {
     }
 }
 
-pub struct VMOutputState {
-    pub registers: Registers,
```
Comment (ludamad) [inline brillig_vm/src/lib.rs:304]: Unused

File: brillig_vm/src/registers.rs [hunk 3]
```diff
@@ -1,42 +1,23 @@
...
-/// Registers will store field element values during the
-/// duration of the execution of the bytecode.
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct Registers {
-    pub inner: Vec<Value>,
-}
-
-impl IntoIterator for Registers {
-    type Item = Value;
-    type IntoIter = RegistersIntoIterator;
```
Comment (ludamad) [inline brillig_vm/src/registers.rs:13]: Unused. I don't think anyone needs to iterate register space?

Comment (kevaundray) [reply brillig_vm/src/registers.rs:13]: Right, I think if it's not used internally then it's not needed

File: brillig_vm/src/registers.rs [hunk 4]
```diff
@@ -1,42 +1,23 @@
...
-        RegistersIntoIterator { registers: self, index: 0 }
-    }
-}
-pub struct RegistersIntoIterator {
-    registers: Registers,
-    index: usize,
+    // Keep track of registers as an option to represent uninitialized registers
+    // These occur when we set a register value at a noncontiguous index, past currently defined registers
+    // This could just store 0's, but it would be
+    pub inner: Vec<Option<Value>>,
```
Comment (dbanks12) [inline brillig_vm/src/registers.rs:9]: Unfinished comment? Also this is still a bit confusing to me.... Can't follow why you'd need/want this.

File: brillig_vm/src/registers.rs (hunk 3 above)
Comment (vezenovm) [reply brillig_vm/src/registers.rs:13]: A register iterator was used previously when setting the brillig outputs in the BrilligSolver, but that has changed now and we can remove this

File: brillig_vm/src/lib.rs [hunk 5]
```diff
@@ -43,22 +43,28 @@ pub struct ForeignCallResult {
...
+    // Instruction pointer
     program_counter: usize,
-    /// A counter maintained throughout a Brillig process that determines
-    /// whether the caller has resolved the results of a [foreign call][Opcode::ForeignCall].
+    // Tracks foreign calls
     foreign_call_counter: usize,
-    /// Represents the outputs of all foreign calls during a Brillig process
-    /// List is appended onto by the caller upon reaching a [VMStatus::ForeignCallWait]
+    // Holds foreign call results
     foreign_call_results: Vec<ForeignCallResult>,
```
Comment (dbanks12) [inline brillig_vm/src/lib.rs:55]: Are these removed comments no longer true? I think they are helpful.

**Review by dbanks12 — APPROVED** [review]
LGTM, couple of minor comments.

File: brillig_vm/src/lib.rs (hunk 5 above)
Comment (vezenovm) [reply brillig_vm/src/lib.rs:55]: I am curious about this removal as well

Comment (ludamad) [reply brillig_vm/src/lib.rs:55]: Just an accident

**Pull request merged**